ingestion_server/ingestion_server/state.py

```diff
@@ -55,31 +57,44 @@ def register_indexing_job(worker_ips, target_index):
         db["worker_statuses"] = worker_statuses
         db["start_time"] = datetime.datetime.now()
         db["target_index"] = target_index
+        db["task_id"] = task_id
         return True
 
 
-def worker_finished(worker_ip):
+def worker_finished(worker_ip, error):
     """
     The scheduler received a notification indicating an indexing worker has
     finished its task.
     :param worker_ip: The private IP of the worker.
-    :return: The target index if all workers are finished, else False.
+    :param error: Whether this worker had an error during processing.
+    :return: A dict containing the target index, task_id, and the percent of
```

Comment:
FWIW it might be worth making this dictionary an explicit `NamedTuple`, that way we can access field attributes easier and have a clear definition of available values/types